packages/core/bootstrap/src/lib/modules/Overrider.ts

```diff
@@ -0,0 +1,122 @@
+import type { CoinsResponse } from '@chainlink/types'
+import { logger } from './logger'
+
+export class Overrider {
+  adapterName: string
+  adapterOverrides: AdapterOverrides
+  internalOverrides: OverrideObj
+  inputOverrides: OverrideObj
+
+  constructor(internalOverrides: unknown, inputOverrides: unknown, adapterName: string) {
+    internalOverrides = internalOverrides || {}
+    inputOverrides = inputOverrides || {}
+    if (!Overrider.isOverrideObj(internalOverrides)) throw Error('Internal overrides are invalid.')
+    if (!Overrider.isOverrideObj(inputOverrides)) throw Error('Input overrides are invalid.')
+    this.internalOverrides = internalOverrides
+    this.inputOverrides = inputOverrides
+    this.adapterName = adapterName.toLowerCase()
+    this.adapterOverrides = this.combineOverrides(
+      internalOverrides?.[this.adapterName] || {},
+      inputOverrides?.[this.adapterName] || {},
+    )
+  }
+
+  performOverrides = (requestedSymbols: RequestedSymbols): [OverriddenCoins, RemainingSymbols] => {
+    const overriddenSyms: OverriddenCoins = {}
+    const remainingSyms: RemainingSymbols = []
+    const reqSyms = Array.isArray(requestedSymbols) ? requestedSymbols : [requestedSymbols]
+    for (const sym of reqSyms) {
+      if (this.adapterOverrides[sym]) overriddenSyms[sym] = this.adapterOverrides[sym]
+      else remainingSyms.push(sym)
+    }
+    return [overriddenSyms, remainingSyms]
+  }
+
+  static convertRemainingSymbolsToIds = (
+    overriddenCoins: OverriddenCoins,
+    remainingSyms: RequestedSymbols,
+    coinsResponse: CoinsResponse[],
+  ): RequestedCoins => {
+    const requestedCoins = overriddenCoins || {}
```

Comment:
This is personal preference, but I'd prefer a default argument rather than having 2 names.
```
    overriddenCoins: OverriddenCoins = {},
    ```